spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/RewriteManifestsProcedure.java

```diff
@@ -118,4 +126,15 @@ private InternalRow[] toOutputRows(RewriteManifests.Result result) {
   public String description() {
     return "RewriteManifestsProcedure";
   }
+
+  private RewriteManifestsSparkAction checkAndApplyFilter(
+      RewriteManifestsSparkAction action, String where, Identifier ident) {
+    if (where != null) {
+      Expression expression = filterExpression(ident, where);
```

Comment:
The reason why we did not add it in the first place is because we persist only lower/upper boundaries for manifest partition values. Therefore, filters may not necessarily benefit from that. Based on what I've seen, these boundaries are rarely helpful when the manifest compaction is needed. Also, we have done some improvements to how the manifests are rewritten (not released yet). From what I see, it is a matter of a few minutes to rewrite all manifests even for large tables because the operation is distributed.

@bknbkn, did you have a chance to try the improvements for the action in master? How long does the rewrite take in your tables if you target all manifests?

Reply:
In our scenario, `Maintenance Action` more often exists as a periodic task, we will automatically call `Maintenance Action` regularly to maintain many existing tables in the warehouse, most of tables are time-partitioned tables. Adding filter brings the following benefits:

- **Reduce unnecessary manifests IO in the cluster**

Within an action execution interval, only a part of the partition data of each table will be updated (in our scenario about 10% of partitions will be affected, the manifests contained remaining partitions have been aggregated when the previous action was executed, and no need to re-aggregate). For a single table, it will not take more than **10 minutes** to execute rewrite manifests in a distributed situation, but we still hope to have the opportunity to reduce the overall pressure on the cluster.

- **Conflict avoidance**

When too many small files are written to the upstream of the Iceberg table at one time, it may trigger manfests merging(By ManifestMergeManager.mergeGroup). This behavior may conflict with the rewrite manifests action. Facing streaming tasks, we cannot stop it, can only be solved by retrying the action. But filter out the partition the task is currently writing can reduce the occurrence of conflicts.

**By the way,** In V1 table with snapshot-id inherit=true, it still need write all manifests in driver when execute copyManifests function, I find that this propertity is no longer affects V2 table, Is any plan to remove or default true this propertity in the V1 table, or is there still any unsafe?

Reply:
I agree that there are some cases where it could be beneficial, but like you said, it would only work in some special cases (e.g. time-based partitioning and writing data only to new partitions and never to old ones). However, I am not sure how generic this is. If we add this new argument, it may not work in a lot of cases, causing confusion.

What do you think about using the underlying action directly for your needs? There, you can pass a closure and have sophisticated filtering logic even beyond partition bounds (e.g. you can check snapshot IDs).

> By the way, In V1 table with snapshot-id inherit=true, it still need write all manifests in driver when execute copyManifests function, I find that this propertity is no longer affects V2 table, Is any plan to remove or default true this propertity in the V1 table, or is there still any unsafe?

I highly recommend enabling that property in all tables if you know that it will not brake some old writers/readers. We can't make that guarantee on behalf of the user.

Reply:
Got it, thanks for your answer @aokolnychyi !

Reply:
time-based partition table is not merely some special cases in our practice. in fact they are the majority cases.

Reply:
> The reason why we did not add it in the first place is because we persist only lower/upper boundaries for manifest partition values. Therefore, filters may not necessarily benefit from that. Based on what I've seen, these boundaries are rarely helpful when the manifest compaction is needed. Also, we have done some improvements to how the manifests are rewritten (not released yet). From what I see, it is a matter of a few minutes to rewrite all manifests even for large tables because the operation is distributed.
> 
> @bknbkn, did you have a chance to try the improvements for the action in master? How long does the rewrite take in your tables if you target all manifests?

could you please provide some pr which are improvements to how the manifests are rewritten? 
We have some big tables which have too much manifest, and conflicts are always happening when rewriting manifest,